refinery/data_set_manager/tests.py

```diff
@@ -2087,6 +2089,58 @@ def test_prepare_node_s3_file_store_item_source_with_datafile(self):
             expected_download_url=self.file_store_item.get_datafile_url()
         )
 
+    def _create_analysis_node_connection(self, direction=OUTPUT_CONNECTION,
+                                         is_refinery_file=True):
+        user = User.objects.create_user("test", "", "test")
+        make_analyses_with_single_dataset(1, user)
+
+        AnalysisNodeConnection.objects.create(
+            analysis=Analysis.objects.first(),
+            node=self.node,
+            direction=direction,
+            step=1,
+            name="{} Analysis Node Connection".format(direction),
+            filename="test.txt",
+            is_refinery_file=is_refinery_file
+        )
+
+    def test_prepare_node_with_non_exposed_input_node_connection_isnt_skipped(
+            self
+    ):
+        self._create_analysis_node_connection(
+            direction=INPUT_CONNECTION,
+            is_refinery_file=False
+        )
+        self._assert_valid_node_index_preparation(
+            self.node,
+            expected_download_url=self.file_store_item.get_datafile_url()
+        )
+
+    def test_prepare_node_with_exposed_input_node_connection_isnt_skipped(
+            self
+    ):
+        self._create_analysis_node_connection(direction=INPUT_CONNECTION)
+        self._assert_valid_node_index_preparation(
+            self.node,
+            expected_download_url=self.file_store_item.get_datafile_url()
+        )
+
+    def test_prepare_node_with_non_exposed_output_node_connection_is_skipped(
+            self
+    ):
+        self._create_analysis_node_connection(is_refinery_file=False)
+        with self.assertRaises(SkipDocument):
```

Comment:
talked with @mccalluc  about this. Odd to have an assertion underneath this with